daisy_workflows/linux_common/utils/common.py

```diff
@@ -88,13 +88,12 @@ def YumInstall(package_list):
 
 @RetryOnFailure()
 def AptGetInstall(package_list, suite=None):
-  if AptGetInstall.first_run:
-    try:
-      Execute(['apt-get', 'update'])
-    except subprocess.CalledProcessError as error:
-      logging.info('Apt update failed, trying again: %s' % error)
-      Execute(['apt-get', 'update'], raise_errors=False)
-    AptGetInstall.first_run = False
+  # When `apt-get update` fails to update a repo, it returns 0.
+  # This check ensures that we retry running update until we've
+  # had one successful install.
+  # https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=778357
+  if not AptGetInstall.prior_success:
+    Execute(['apt-get', 'update'])
```

Comment:
Should there be some kind of timeout/re-try count limit? Daisy step will timeout eventually but if apt-get is not expected to succeed after 10s of minutes of re-tries anyway, it might make sense to limit them.

Reply:
Here are the defaults for the retry annotation: 

https://github.com/GoogleCloudPlatform/compute-image-tools/blob/2519227884218345e99c81207db99eaae86dd0e5/daisy_workflows/linux_common/utils/common.py#L36